okhttp/src/main/java/io/grpc/okhttp/OkHttpServerTransport.java

```diff
@@ -161,10 +161,14 @@ public void start(ServerTransportListener listener) {
 
   private void startIo(SerializingExecutor serializingExecutor) {
     try {
-      bareSocket.setTcpNoDelay(true);
+      synchronized (lock) {
```

Comment:
Do we still need this synchronized? If we do, we're going to need a comment as to why (probably a link to the Java bug).

Reply:
Yes--and just added the comment.

Reply:
Let's explain a bit more.

> The socket implementation is lazily initialized, but had broken thread-safety for that laziness https://bugs.openjdk.org/browse/JDK-8278326 . As a workaround, we lock to synchronize initialization with shutdown().